src/collections/componentSet.ts

```diff
@@ -292,6 +295,20 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
 
     const mdapiDeploy = new MetadataApiDeploy(operationOptions);
     await mdapiDeploy.start();
+
+    // Creates an array of unique metadata types to be deployed, uses Set to avoid duplicates.
+    const listOfMetadataTypesDeployed = Array.from(new Set(toDeploy.map((c) => c.type.name)));
+
+    void Lifecycle.getInstance().emitTelemetry({
+      eventName: 'metadata_api_deploy',
+      library: 'SDR',
+      function: 'ComponentSet.deploy',
+      totalNumberOfComponents: this.size,
+      numberOfComponentsDeployed: toDeploy.length,
+      componentsDeployed: listOfMetadataTypesDeployed.toString(),
```

Comment:
We should be careful about the names of these fields.  Nothing has been deployed yet.  Everything at this point is, "to be deployed", or "the metadata being requested for deployment".
`componentsDeployed` as a name can't be correct since you're providing the unique metadata types, not the components.

Reply:
I'll change it.

@shetzel I pulled this naming directly from the PR where this was removed from toolbelt.
https://github.com/salesforcecli/toolbelt/pull/215/files#diff-ca446f5ccf231e22c85434264adcb8d977f76d6375c1ae789fd39c1bde5aa324L329